staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -711,3 +871,17 @@ func isTooLargeResourceVersionError(err error) bool {
 
 	return false
 }
+
+// isWatchErrorRetriable determines if it is safe to retry
```

Comment:
This is also refactoring - let's move to a separate PR (same as mentioned in another comment) to make this PR as simple ss possible.